scripts/tasks.sh

```diff
@@ -640,12 +641,75 @@ EOF
-        return 0
+    else
+        bail "sync-object-store pod failed"
+    fi
+
+    # Update kotsadm to use minio
+    if kubernetes_resource_exists default secret kotsadm-s3; then
+        echo "Updating kotsadm to use minio"
+        kubectl patch secret kotsadm-s3 -p "{\"stringData\":{\"access-key-id\":\"${MINIO_ACCESS_KEY_ID}\",\"secret-access-key\":\"${MINIO_ACCESS_KEY_SECRET}\",\"endpoint\":\"http://${MINIO_HOST}\",\"object-store-cluster-ip\":\"${MINIO_CLUSTER_IP}\"}}"
+
+        if kubernetes_resource_exists default deployment kotsadm; then
+            kubectl rollout restart deployment kotsadm
+        elif kubernetes_resource_exists default statefulset kotsadm; then
+            kubectl rollout restart statefulset kotsadm
+        fi
+    fi
+
+    # Update registry to use minio
+    if kubernetes_resource_exists kurl configmap registry-config; then
+        echo "Updating registry to use minio"
+        kubectl -n kurl get configmap registry-config -ojsonpath='{ .data.config\.yml }' | sed "s/regionendpoint: http.*/regionendpoint: http:\/\/${MINIO_CLUSTER_IP}/" > config.yml
+        kubectl -n kurl delete configmap registry-config
+        kubectl -n kurl create configmap registry-config --from-file=config.yml=config.yml
+        rm config.yml
+    fi
+    if kubernetes_resource_exists kurl secret registry-s3-secret; then
+        kubectl -n kurl patch secret registry-s3-secret -p "{\"stringData\":{\"access-key-id\":\"${MINIO_ACCESS_KEY_ID}\",\"secret-access-key\":\"${MINIO_ACCESS_KEY_SECRET}\",\"object-store-cluster-ip\":\"${MINIO_CLUSTER_IP}\"}}"
+    fi
+    if kubernetes_resource_exists kurl deployment registry; then
+        kubectl -n kurl rollout restart deployment registry
+    fi
+
+    # Update velero to use minio only if currently using RGW since velero may have already been
+    # updated to use an off-cluster object store.
+    if kubernetes_resource_exists velero backupstoragelocation default; then
+        echo "Updating velero to use minio"
+        s3Url=$(kubectl -n velero get backupstoragelocation default -ojsonpath='{ .spec.config.s3Url }')
+        if [ "$s3Url" = "http://rook-ceph-rgw-rook-ceph-store.rook-ceph" ]; then
+            kubectl -n velero patch backupstoragelocation default --type=merge -p "{\"spec\":{\"config\":{\"s3Url\":\"http://${MINIO_HOST}\",\"publicUrl\":\"http://${MINIO_CLUSTER_IP}\"}}}"
+
+            resticrepo=$(kubectl -n velero get resticrepositories --selector=velero.io/storage-location=default --no-headers | awk '{ print $1 }')
+            if [ -n "$resticrepo" ]; then
+                kubectl -n velero patch resticrepositories "$resticrepo" --type=merge -p "{\"spec\":{\"resticIdentifier\":\"s3:http://${MINIO_HOST}/velero/restic/default\"}}"
+            fi
+        else
+            echo "default backupstoragelocation was not rgw, skipping"
+        fi
+    fi
+    if kubernetes_resource_exists velero secret cloud-credentials; then
```

Comment:
should we only do this if the storage location was set to rook-ceph?

Reply:
It checks on line 694 that the creds are for RGW before overwriting them. I think it's a bit more resilient to only rely on that check so that if there was some failure after the storage location was updated to minio it will still update the credentials.